datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -425,7 +437,8 @@ impl Stream for ParquetExecStream {
         mut self: Pin<&mut Self>,
         _cx: &mut Context<'_>,
     ) -> Poll<Option<Self::Item>> {
-        Poll::Ready(Iterator::next(&mut *self))
+        let poll = Poll::Ready(Iterator::next(&mut *self));
+        self.baseline_metrics.record_poll(poll)
```

Comment:
I think we remove `async` so will never call poll. But i think we should keep this change.

Reply:
I think the reason poll is not called is that the new scheduler is not (yet) connected up. I agree we should keep this change. 

cc @tustvold